GUI: Allow opening BIF files directly

michaelpm54 opened the pull request:
Let me know if you want any changes of course or if it should be done a different way.

DrMcCoy commented on src/aurora/keyfile.cpp:
```diff
@@ -202,4 +202,18 @@ Common::SeekableReadStream *KEYFile::getResource(uint32 index, bool UNUSED(tryNo
 	return iRes.dataFile->getResource(iRes.resIndex);
 }
 
+std::vector<Archive::Resource *> KEYFile::getResourceListForDataFile(Common::UString dataFile) {
```
That should take a `const Common::UString &`.

And the return could be a `std::vector<const Archive::Resource *>`

The method itself is const, too.

DrMcCoy commented on src/gui/resourcetree.cpp:
```diff
@@ -205,15 +225,23 @@ bool ResourceTree::hasChildren(const QModelIndex &index) const {
 }
 
-void ResourceTree::insertItemsFromArchive(Archive &archive, const QString &path,
+void ResourceTree::insertItemsFromArchive(Archive &archive, ResourceTreeItem *item,
```
That can be a `const ResourceTreeItem &`, no?

michaelpm54 commented:
Updated.
- Added const where requested and replaced pointer with ref.
- Added an early return [here](https://github.com/xoreos/phaethon/pull/30/commits/1b891616752928febdd8c91686fe14b1babc128b#diff-d586d5c0a96afc76f19d4078c059e6e8R197) which should prevent adding items twice.

DrMcCoy commented:
You forgot making `KEYFile::getResourceListForDataFile()` itself const.

michaelpm54 commented:
Good eye, fixed.

DrMcCoy commented:
Merged, thanks! :)
